loghandler.go

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2021 Mia srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package glogger
+
+import (
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+type LoggerHandler struct {
+	Logger         *logrus.Logger
+	ExcludedPrefix []string
+	OnNext         func(rrw *ReadableResponseWriter, httpRequest *http.Request)
+}
+
+func HandleLog(loggerHandler LoggerHandler) func(w http.ResponseWriter, r *http.Request) {
```

Comment:
This function could be internal, so lower case?
Because the function returns a function `func(w http.ResponseWriter, r *http.Request)`, so it could not be reused for other router.

But in this way, it is not so much reusable

Reply:
My idea is to use this to create an ad-hoc middleware for echo.
In the echo context, using context.Response().Writer, you take the response writer, and in a similar way also the request.

The usage for echo should be similar to:

```
// inside the echo middleware
loggerHandler := LoggerHandler{
     ..., 
     OnNext: func(...) {
       echoContext.Next(....)
     }
}

HandleLog(loggerHandler)(echoContext.Response().Writer, echoContext.Request()....)

```